queue_job/jobrunner/runner.py

```diff
@@ -378,6 +378,44 @@ def _query_requeue_dead_jobs(self):
             RETURNING uuid
             """
 
+    def _query_requeue_orphaned_jobs(self):
+        """Query to requeue jobs stuck in 'enqueued' state without a lock.
+
+        This handles the edge case where the runner marks a job as 'enqueued'
+        but the HTTP request to start the job never reaches the Odoo server
+        (e.g., due to server shutdown/crash between setting enqueued and
+        the controller receiving the request). These jobs have no lock record
+        because set_started() was never called, so they are invisible to
+        _query_requeue_dead_jobs().
+        """
+        return """
+            UPDATE
+                queue_job
+            SET
+                state='pending'
+            WHERE
+                id in (
+                    SELECT
+                        id
+                    FROM
+                        queue_job
+                    WHERE
+                        state = 'enqueued'
+                        AND NOT EXISTS (
+                            SELECT
+                                1
+                            FROM
+                                queue_job_lock
+                            WHERE
+                                queue_job_id = queue_job.id
+                        )
+                        AND date_enqueued <
+                            (now() AT TIME ZONE 'utc' - INTERVAL '10 sec')
+                    FOR UPDATE SKIP LOCKED
+                )
```

Comment:
```suggestion
            SELECT uuid FROM queue_job
            WHERE
                state = 'enqueued'
                AND date_enqueued < (now() AT TIME ZONE 'utc' - INTERVAL '10 sec')
                AND NOT EXISTS (
                    SELECT
                        1
                    FROM
                        queue_job_lock
                    WHERE
                        queue_job_id = queue_job.id
                )
```

Reply:
@sbidoul I've updated the query and add a test case for this